diff --git a/CMakeLists.txt b/CMakeLists.txt
index acd597d..67d654d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -170,9 +170,21 @@ if(CMAKE_C_COMPILER_ID MATCHES "MSVC" OR CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
     # lots of warnings with cl.exe right now, use /W1
     add_definitions("/W1 -D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS /bigobj")
 else()
-    add_definitions( "-Wall -Wno-unknown-pragmas -Wno-unused-function -Wa,-mbig-obj" )
+    add_definitions( "-Wall -Wno-unknown-pragmas -Wno-unused-function" )
 endif()
 
+if (CMAKE_SYSTEM_NAME MATCHES "Windows")
+   if ((CMAKE_GENERATOR MATCHES "MinGW Makefiles") OR
+       (CMAKE_GENERATOR MATCHES "MSYS Makefiles") OR
+       (CMAKE_GENERATOR MATCHES "Unix Makefiles"))
+
+      set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -Og")
+
+      if (CMAKE_BUILD_TYPE STREQUAL  "")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O1")
+      endif()
+   endif()
+endif()
 
 # install and export variables
 set(config_install_dir "lib/cmake/${PROJECT_NAME}")
